pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,512 @@
+	t.Cleanup(MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		item             pam.Item
+		value            *string
+		expectedGetError error
+		expectedSetError error
+		expectedValue    *string
+	}{
+		"set invalid item": {
+			item:             pam.Item(-1),
+			value:            ptrValue("some value"),
+			expectedSetError: pam.BadItem,
+		},
+
+		"get invalid item": {
+			item:             pam.Item(0),
+			expectedGetError: pam.BadItem,
+			expectedValue:    ptrValue(""),
+		},
+
+		"set user": {
+			item:  pam.User,
+			value: ptrValue("an user"),
+		},
+
+		"get unset user": {
+			item:          pam.User,
+			expectedValue: ptrValue(""),
+		},
+
+		"get and set user": {
+			item:          pam.User,
+			value:         ptrValue("the-user"),
+			expectedValue: ptrValue("the-user"),
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
+
+			require.True(t, tc.value != nil || tc.expectedValue != nil)
+
+			tx := NewModuleTransactionDummy(nil)
+
+			if tc.value != nil {
+				err := tx.SetItem(tc.item, *tc.value)
+				require.ErrorIs(t, err, tc.expectedSetError)
+			}
+
+			if tc.expectedValue != nil {
+				value, err := tx.GetItem(tc.item)
+				require.Equal(t, value, *tc.expectedValue)
+				require.ErrorIs(t, err, tc.expectedGetError)
+			}
+		})
+	}
+}
+
+func TestSetPutEnv(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		env              string
+		value            *string
+		presetValues     map[string]string
+		skipPut          bool
+		expectedPutError error
+		expectedValue    *string
+	}{
+		"put var": {
+			env:   "AN_ENV",
+			value: ptrValue("value"),
+		},
+
+		"unset non-preset var": {
+			env:           "NEVER_SET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"unset preset var": {
+			presetValues:  map[string]string{"PRESET_ENV": "hey!"},
+			env:           "PRESET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"change preset var": {
+			presetValues:  map[string]string{"PRESET_ENV": "hey!"},
+			env:           "PRESET_ENV",
+			value:         ptrValue("hello!"),
+			expectedValue: ptrValue("hello!"),
+		},
+
+		"get unset env": {
+			skipPut:       true,
+			env:           "AN_UNSET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"put invalid env": {
+			env:              "",
+			value:            ptrValue("Invalid Value"),
+			expectedPutError: pam.BadItem,
+		},
+
+		"get invalid env": {
+			env:           "",
+			value:         ptrValue("Invalid Value"),
+			expectedValue: ptrValue(""),
+			skipPut:       true,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
+
+			require.False(t, tc.skipPut && tc.expectedValue == nil)
+
+			tx := NewModuleTransactionDummy(nil)
+			envList, err := tx.GetEnvList()
+			require.NoError(t, err)
+			require.Len(t, envList, 0)
```

Comment:
As those are setup assertions, I would add a message here (as an argument of the `require` calls) to help when debugging test failure, like: `Setup: GetEnvList should not return an error". This way, by reading the test report (in case of failure) we can already triage where the problem is.